lib/logstash/outputs/kafka.rb

```diff
@@ -20,132 +20,114 @@
-  # Provides a way to specify a partition key as a string. To specify a partition key for
-  # Kafka, configure a format that will produce the key as a string. Defaults
-  # `key_serializer_class` to `kafka.serializer.StringEncoder` to match. For example, to partition
-  # by host:
-  # [source,ruby]
-  #     output {
-  #       kafka {
-  #           partition_key_format => "%{host}"
-  #       }
-  #     }
-  config :partition_key_format, :validate => :string, :default => nil
 
   public
   def register
     LogStash::Logger.setup_log4j(@logger)
 
     options = {
-        :broker_list => @broker_list,
-        :compression_codec => @compression_codec,
-        :compressed_topics => @compressed_topics,
-        :request_required_acks => @request_required_acks,
-        :serializer_class => @serializer_class,
-        :partitioner_class => @partitioner_class,
-        :request_timeout_ms => @request_timeout_ms,
-        :producer_type => @producer_type,
-        :key_serializer_class => @key_serializer_class,
-        :message_send_max_retries => @message_send_max_retries,
-        :retry_backoff_ms => @retry_backoff_ms,
-        :topic_metadata_refresh_interval_ms => @topic_metadata_refresh_interval_ms,
-        :queue_buffering_max_ms => @queue_buffering_max_ms,
-        :queue_buffering_max_messages => @queue_buffering_max_messages,
-        :queue_enqueue_timeout_ms => @queue_enqueue_timeout_ms,
-        :batch_num_messages => @batch_num_messages,
-        :send_buffer_bytes => @send_buffer_bytes,
-        :client_id => @client_id
+      :key_serializer => @key_serializer,
+      :value_serializer => @value_serializer,
+      :bootstrap_servers => @bootstrap_servers,
+      :acks => @acks,
+      :buffer_memory => @buffer_memory,
+      :compression_type => @compression_type,
+      :retries => @retries,
+      :batch_size => @batch_size,
+      :client_id => @client_id,
+      :linger_ms => @linger_ms,
+      :max_request_size => @max_request_size,
+      :receive_buffer_bytes => @receive_buffer_bytes,
+      :send_buffer_bytes => @send_buffer_bytes,
+      :timeout_ms => @timeout_ms,
+      :block_on_buffer_full => @block_on_buffer_full,
+      :metadata_fetch_timeout_ms => @metadata_fetch_timeout_ms,
+      :metadata_max_age_ms => @metadata_max_age_ms,
+      :reconnect_backoff_ms => @reconnect_backoff_ms,
+      :retry_backoff_ms => @retry_backoff_ms
     }
-    @producer = Kafka::Producer.new(options)
+    @producer = Kafka::KafkaProducer.new(options)
     @producer.connect
 
-    @logger.info('Registering kafka producer', :topic_id => @topic_id, :broker_list => @broker_list)
+    @logger.info('Registering kafka producer', :topic_id => @topic_id, :bootstrap_servers => @bootstrap_servers)
 
     @codec.on_event do |event, data|
       begin
-        @producer.send_msg(event.sprintf(@topic_id),@partition_key,data)
+        @producer.send_msg(event.sprintf(@topic_id), nil, nil, data)
```

Comment:
just curious: we can't have a per message serializer now? i.e. picking it at the time of sending messages?